tensorflow/core/util/gpu_launch_config.h

```diff
@@ -141,24 +141,41 @@ inline CudaLaunchConfig GetCudaLaunchConfig(int work_element_count,
   return config;
 }
 
-// Calculate the Cuda launch config we should use for a kernel launch. This
+// Calculate the Gpu launch config we should use for a kernel launch. This
 // variant takes the resource limits of func into account to maximize occupancy.
 // REQUIRES: work_element_count > 0.
 template <typename DeviceFunc>
-inline CudaLaunchConfig GetCudaLaunchConfig(int work_element_count,
+inline GpuLaunchConfig GetGpuLaunchConfig(int work_element_count,
                                             const Eigen::GpuDevice& d,
                                             DeviceFunc func,
                                             size_t dynamic_shared_memory_size,
                                             int block_size_limit) {
   CHECK_GT(work_element_count, 0);
-  CudaLaunchConfig config;
+  GpuLaunchConfig config;
   int block_count = 0;
   int thread_per_block = 0;
 
+#if GOOGLE_CUDA
   cudaError_t err = cudaOccupancyMaxPotentialBlockSize(
       &block_count, &thread_per_block, func, dynamic_shared_memory_size,
       block_size_limit);
   CHECK_EQ(err, cudaSuccess);
+#elif TENSORFLOW_USE_ROCM
+  // ROCM TODO re-enable this after hipOccupancyMaxPotentialBlockSize is
+  // implemented
+  //hipError_t err = hipOccupancyMaxPotentialBlockSize(
+  //    &block_count, &thread_per_block, func, dynamic_shared_memory_size,
+  //    block_size_limit);
+  //CHECK_EQ(err, hipSuccess);
+
+  const int physical_thread_count = std::min(
+      d.getNumGpuMultiProcessors() * d.maxGpuThreadsPerMultiProcessor(),
+      work_element_count);
+  thread_per_block = std::min(1024, d.maxGpuThreadsPerBlock());
+  block_count =
+      std::min(DivUp(physical_thread_count, thread_per_block),
+               d.getNumGpuMultiProcessors());
```

Comment:
Is the one block per SM limit intentional?

Reply:
I'll need to check with HIP runtime team on this heuristic. It was given by them before we really have `hipOccupancyMaxPotentialBlockSize` implemented in HIP runtime.

Reply:
Turns out the logic is just a clone of `GetGpuLaunchConfig(int, const Eigen::GpuDevice&)` above which assumes the kernel be simple and memory-bounded.

Reply:
Added comment to reflect this.